packages/apollo-server/src/express.ts

```diff
@@ -20,16 +21,17 @@ export class ApolloServer extends ApolloServerBase<
     const { app, request } = config;
 
     app.use(config.path, cors(config.cors), json(), (req, res, next) => {
+      // only show graphiql if we want html for a GET
+      const accept = accepts(req);
+      const types = accept.types() as string[];
+      const isHTML =
+        types.find(
+          (x: string) => x === 'text/html' || x === 'application/json',
+        ) === 'text/html';
+
       // make sure we check to see if graphiql should be on
       // change opts.graphiql type to be boolean
-      if (
-        config.graphiql &&
-        req.method === 'GET' &&
-        req.headers['content-type'] !== 'application/json' &&
-        req.params &&
-        !req.params.query
-      ) {
-        // want to return graphiql
+      if (config.graphiql !== false && req.method === 'GET' && isHTML) {
```

Comment:
Do we want to keep calling this configuration option `graphiql`?  I think this is the first version of the server where we've introduced the option, so now is our change to change it.  For example, to `gui`.